src/Markdown.js

```diff
@@ -56,23 +57,47 @@ export default class Markdown {
         return is_plain;
     }
 
-    toHTML() {
+    render(html) {
         const parser = new commonmark.Parser();
 
         const renderer = new commonmark.HtmlRenderer({safe: true});
         const real_paragraph = renderer.paragraph;
-        renderer.paragraph = function(node, entering) {
-            // If there is only one top level node, just return the
-            // bare text: it's a single line of text and so should be
-            // 'inline', rather than unnecessarily wrapped in its own
-            // p tag. If, however, we have multiple nodes, each gets
-            // its own p tag to keep them as separate paragraphs.
-            var par = node;
-            while (par.parent) {
-                par = par.parent
+        if (html) {
+            renderer.paragraph = function(node, entering) {
+                // If there is only one top level node, just return the
+                // bare text: it's a single line of text and so should be
+                // 'inline', rather than unnecessarily wrapped in its own
+                // p tag. If, however, we have multiple nodes, each gets
+                // its own p tag to keep them as separate paragraphs.
+                var par = node;
+                while (par.parent) {
+                    par = par.parent
+                }
+                if (par.firstChild != par.lastChild) {
+                    real_paragraph.call(this, node, entering);
+                }
             }
-            if (par.firstChild != par.lastChild) {
-                real_paragraph.call(this, node, entering);
+        } else {
+            renderer.out = function(s) {
+                this.lit(s);
```

Comment:
Ah, seems I only added the comment in the commit message.  Will fix that.  The normal `out` function runs the output through a sanitizer that will entity encode things like quotation marks, which is problematic when it happens multiple times.